runtime/commonTest/src/kotlinx/serialization/json/JsonInputOutputRecursiveTest.kt

```diff
@@ -155,49 +164,52 @@ class JsonInputOutputRecursiveTest : JsonTestBase() {
     @Serializable
     private data class Event(
         val id: Int,
-        @Serializable(with = EitherSerializer::class) val payload: DummyEither,
+        @Serializable(with = EitherSerializer::class) val payload: Either,
         val timestamp: Long
     )
 
     @Serializable(with = RecursiveSerializer::class)
-    private sealed class DummyRecursive {
+    private sealed class SealedRecursive {
         @Serializable
-        data class A(val value: Int): DummyRecursive()
+        data class A(val value: Int) : SealedRecursive()
 
         @Serializable
-        data class B(val children: List<DummyRecursive>): DummyRecursive()
+        data class B(val children: List<SealedRecursive>) : SealedRecursive()
     }
 
-    private object RecursiveSerializer: KSerializer<DummyRecursive> {
+    private object RecursiveSerializer : KSerializer<SealedRecursive> {
         private const val typeAttribute = "type"
         private const val typeNameA = "a"
         private const val typeNameB = "b"
 
-        override val descriptor: SerialDescriptor
-            get() = SerialClassDescImpl("DummyRecursive")
+        // TODO in builder is not suitable for recursive descriptors
```

Comment:
well, this might be a problem. do you have ideas how can we overcome that?

Reply:
Already done in the next commit

Reply:
not clear how `defer {}` helps here